keyboards/elephant42/elephant42.c

```diff
@@ -0,0 +1,66 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "elephant42.h"
+
+#if defined(RGB_MATRIX_ENABLE)
+led_config_t g_led_config = {
+    { // keymap to LED number
+        // left
+        {    19,     18, 13, 12, 7, 6}, // R3
+        {    20,     17, 14, 11, 8, 5}, // R2
+        {NO_LED,     16, 15, 10, 9, 4}, // R1
+        {NO_LED, NO_LED,  0,  1, 2, 3}, // R0
+        // right
+        {    46,     45, 40, 39, 34, 33}, // R3
+        {    47,     44, 41, 38, 35, 32}, // R2
+        {NO_LED,     43, 42, 37, 36, 31}, // R1
+        {NO_LED, NO_LED, 27, 28, 29, 30}, // R0
+    },
+    {// phisical LED position, {x, y} : x = 0..224, y = 0..64
+        // left-bg [LED_01-21]
+        {56, 65}, {77, 68}, {95, 79}, {109, 91}, 
+        {87, 46}, {87, 29}, {87, 12}, { 69, 43}, { 69, 26}, { 69,  8}, 
+        {52, 35}, {52, 17}, {52,  0}, { 35,  4}, { 35, 21}, { 35, 38}, 
+        {17, 54}, {17, 36}, {17, 19}, {  0, 26}, {  0, 43}, 
+        // left-ug [LED_22-27]
+        {25, 49}, {26, 19}, {60, 14}, {106, 33}, {107, 72}, { 70, 66}, 
+        // right-bg [LED_28-48]
+        {199, 65}, {179, 68}, {160, 79}, {146, 91}, 
+        {168, 46}, {168, 29}, {168, 12}, {186, 43}, {186, 26}, {186,  8}, 
+        {203, 35}, {203, 17}, {203,  0}, {220,  4}, {220, 21}, {220, 38}, 
+        {238, 54}, {238, 36}, {238, 19}, {255, 26}, {255, 43}, 
+        // right-ug [LED_49-54]
+        {230, 49}, {230, 19}, {195, 14}, {149, 33}, {148, 72}, {185, 66}, 
+    },
+    {// LED Index to Flag
+        // Left-bg [LED_01-21]
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        // left-ug [LED_22-27]
+        LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW,
+        // right-bg [LED_28-48]
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        // right-ug [LED_49-54]
+        LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW, LED_FLAG_UNDERGLOW
+    }
+};
+#endif
```

Comment:
```suggestion
#endif

#ifdef OLED_ENABLE
#   define STEPS 32

uint8_t keypresses = 0;

bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
    if (record->event.pressed) {
        keypresses = (keypresses + 1) % STEPS;
    }
    return process_record_user(keycode, record);
}

bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
    if (is_keyboard_master()) {
        static char layer_names[NUMBER_OF_LAYERS][10] = {"Default", "Lower", "Raise", "Adjust"};
        static char l1[] = "                \x94\x95\x96\x97";
        static char l2[] = "                \xB4\xB5\xB6\xB7";
        static char r1[] = "                \x98\x99\x9A\x9B";
        static char r2[] = "                \xB8\xB9\xBA\xBB";
        int iconShift = keypresses % STEPS;
        if (iconShift > STEPS / 2) {
            iconShift = STEPS - iconShift;
        }
        bool goingLeft = keypresses < STEPS / 2;

        oled_write_P(PSTR("Layer: "), false);
        oled_write_ln(layer_names[get_highest_layer(layer_state)], false);
        oled_advance_page(true);
        oled_write_ln((goingLeft ? l1 : r1) + iconShift, false);
        oled_write_ln((goingLeft ? l2 : r2) + iconShift, false);
    } else {
        oled_advance_page(true);
        static const char PROGMEM logo[] = {
            0x8f,0x90,0x91,0x92,0x93,0x80,0x81,0x82,0x83,0x84,0x85,0x86,0x87,0x88,0x89,0x8a,0x8b,0x8c,0x8d,0x8e,0x0a,
            0xaf,0xb0,0xb1,0xb2,0xb3,0xa0,0xa1,0xa2,0xa3,0xa4,0xa5,0xa6,0xa7,0xa8,0xa9,0xaa,0xab,0xac,0xad,0xae,0x0a,
            0xcf,0xd0,0xd1,0xd2,0xd3,0xc0,0xc1,0xc2,0xc3,0xc4,0xc5,0xc6,0xc7,0xc8,0xc9,0xca,0xcb,0xcc,0xcd,0xce,0
        };

        oled_write_ln_P(logo, false);
    }

    return true;
}
#endif
```

Reply:
Hmmmm, sorry, not sure what would a proper way to define it :(
I mean there is a reference to `NUMBER_OF_LAYERS` (`oled_task_kb` depends on the number of layers) and different keymaps can have different number of layers 🤔

Reply:
well, might be worth just using 4 for that. the default keymap uses 3, and the via keymap uses 4.  So just setting them may be fine.   And if a user needs to change them, it can be done simply enough.